week4/src/components/SearchSection/index.tsx

```diff
@@ -0,0 +1,97 @@
+import { KAKAO } from "core/api";
+import { ResultList, ResultListWithAxios } from "core/types";
+import { useRef, useState } from "react";
+import styled from "styled-components";
+import getLocation from "util/getLocation";
+
+import InputInformation from "./InputInformation";
+
+interface SearchSectionProps {
+  handleResultList: (newResultList: ResultList[]) => void;
+  handleIsLoading: (_isLoading: boolean) => void;
+}
+
+export default function SearchSection(props: SearchSectionProps) {
+  const { handleResultList, handleIsLoading } = props;
+
+  const [isRegionBasedChecked, setIsRegionBasedChecked] = useState<boolean>(false);
+  const locationInputRef = useRef<HTMLInputElement>(null);
+
+  const toggleIsRegionBasedChecked = () => {
+    setIsRegionBasedChecked((prev) => !prev);
+  };
+
+  const 특정지역맥주집가져오기 = async (location = "") => {
+    handleIsLoading(true);
+
+    const {
+      data: { documents },
+    } = await KAKAO.get<ResultListWithAxios>("/search/keyword", {
+      params: {
+        query: location + " " + "맥주",
+      },
+    });
+
+    handleResultList(documents);
+    handleIsLoading(false);
+  };
+
+  const 내근처맥주집가져오기 = async () => {
+    handleIsLoading(true);
+
+    const myLocation = await getLocation();
+
+    if (myLocation) {
+      const {
+        data: { documents },
+      } = await KAKAO.get<ResultListWithAxios>("/search/keyword", {
+        params: {
+          x: myLocation.x,
+          y: myLocation.y,
+          radius: 1000,
+          query: "맥주",
+        },
+      });
+
+      handleResultList(documents);
+    }
+
+    handleIsLoading(false);
+  };
+
+  const 검색하기 = () => {
+    if (isRegionBasedChecked) 내근처맥주집가져오기();
+    else 특정지역맥주집가져오기(locationInputRef.current?.value);
+  };
```

Comment:
파람쓰 구하는 부분 제외하고는 두 함수가 똑같네요!
그래서 파람쓰 구하는 함수를 따로 만들고, 거기서 isRegionBasedChecked 변수 갖고 다른 파람쓰를 가져다 주면 
요청 보내고 setState하는 부분은 같이 쓰면 좋을 것 같습니다

```suggestion
  const 검색하기 = async () => {
    handleIsLoading(true);
    const 쿼리파람쓰 = await getQueryParams(isRegionBasedChecked); // getLocation 때문에 async 함수가 됨
    const {
      data: { documents },
    } = await KAKAO.get<ResultListWithAxios>("/search/keyword", {
      params: 쿼리파람쓰,
    });

    handleResultList(documents);
    handleIsLoading(false);
  };
```

Reply:
omg .. . .